src/site/content/en/blog/responsiveness/index.md

```diff
@@ -0,0 +1,304 @@
+      <th>Latency</th>
+    </tr>
+    <tr>
+      <td>Click</td>
+      <td>120 ms</td>
+    </tr>
+    <tr>
+      <td>Click</td>
+      <td>20 ms</td>
+    </tr>
+    <tr>
+      <td>Key press</td>
+      <td>60 ms</td>
+    </tr>
+    <tr>
+      <td>Key press</td>
+      <td>80 ms</td>
+    </tr>
+  </table>
+</div>
+
+#### Worst interaction latency
+
+The largest, individual interaction latency that occurred on a page. Given the
+example interactions listed above, the worst interaction latency would be 120
+ms.
+
+#### Budgets strategies
+
+[User experience
+research](https://www.tactuallabs.com/papers/howMuchFasterIsFastEnoughCHI15.pdf)
+suggests that users may not perceive latencies below certain thresholds as
+negative. Based on this research we're considering several budget strategies
+using on the following thresholds for each event type:
+
+<div class="w-table-wrapper">
+  <table>
+    <tr>
+      <th>Interaction type</th>
+      <th>Budget threshold</th>
+    </tr>
+    <tr>
+      <td>Click/tap</td>
+      <td>100 ms</td>
+    </tr>
+    <tr>
+      <td>Drag</td>
+      <td>100 ms</td>
+    </tr>
+    <tr>
+      <td>Keyboard</td>
+      <td>50 ms</td>
+    </tr>
+  </table>
+</div>
+
+Each of these strategies will only consider the latency that is more than the
+budget threshold per interaction. Using the example page visit above, the
+over-budget amounts would be as follows:
+
+<div class="w-table-wrapper">
+  <table>
+    <tr>
+      <th>Interaction</th>
+      <th>Latency</th>
+      <th>Latency over budget</th>
+    </tr>
+    <tr>
+      <td>Click
+    </td>
+      <td>120 ms</td>
+      <td>20 ms</td>
+    </tr>
+    <tr>
+      <td>Click
+    </td>
+      <td>20 ms</td>
+      <td>0 ms</td>
+    </tr>
+    <tr>
+      <td>Key press
+    </td>
+      <td>60 ms</td>
+      <td>10 ms</td>
+    </tr>
+    <tr>
+      <td>Key press
+    </td>
+      <td>80 ms</td>
+      <td>30 ms</td>
+    </tr>
+  </table>
+</div>
+
+#### Worst interaction latency over budget
+
+The largest single interaction latency over budget. Using the above example, the
+score would be `max(20, 0, 10, 30) = 30 ms`.
+
+#### Total interaction latency over budget
+
+The sum of all interaction latencies over budget. Using the above example, the
+score would be `(20 + 0 + 10 + 30) = 60 ms`.
+
+#### Average interaction latency over budget
+
+The total over-budget interaction latency divided by the total number of
+interactions. Using the above example, the score would be `(20 + 0 + 10 + 30) /
+4 = 15 ms`.
+
+#### High quantile approximation
+
+As an alternative to computing the largest interaction latency over budget, we
+also considered using a high quantile approximation, which should be fairer to
+web pages that have a lot of interactions and may be more likely to have large
+outliers. We've identified two potential high-quantile approximation strategies
+we like:
+
+- **Option 1:** Keep track of the largest interaction over budget. After every
```

Comment:
nit: this option would need to keep track of the two largest interactions over budget.

Reply:
I think it's OK to be a bit handwavy here and omit this implementation detail? Also it wasn't clear to me if the strategy always goes to the next largest interaction (all time) or if it goes to the next largest interaction since the discard—because wouldn't' that require keeping track of more than just the top two?

Reply:
The confusion perhaps proves that this is not clear enough :) The implementation keeps track of two pseudo-largest values. Each time a new value arrives, the worst/second worst are updated as needed. And every 50 interactions the worst is replaced with second worst and the second worst is cleared. Anyways, not sure how much details we need to provide but if we're asking for feedback on the options then they should be clear.

Reply:
OK, so just to make sure I understand, let's say the first 50 interactions on a page are all terrible (e.g. all between 900-1000ms)). With this option, only the top 2 are kept, right? So then if after 100 more interactions on that page there were never any interactions above the budget threshold, all of those 50 really poor interactions would go away because they would no longer be the first or the second worst, right?

Reply:
Correct, and that is a major concern with this 'approximation'.

Reply:
Ok, I've updated the text to clarify.